### DISCUSSION TITLE
Best way to debug berry script?

### DISCUSSION THREAD

**Discussion by Kugeleis** [arendst/Tasmota#17418] [Q&A] [answered]
I need to find my error in my script. Console says `BRY: Exception> 'type_error' - BRY: argument must be a function`. But without a hint to the code line in my berry script file it's hard to debug. what is the best method to find the error?

**Comment by s-hadinger** [comment, marked as answer]
If you write Berry code, you should compile with `#define USE_BERRY_DEBUG` and you will see line numbers. It is not enabled by default because code takes ~20% more memory

**Comment by joplhb** [comment]
I have also problems, but i think it is the 
d1 = CCS811()
tasmota.add_driver(d1)

The errormessage is gone when i delete it. I can't see any differences to the examples, does anybody see my mistake?

`#- *************************************** -#
#- *************************************** -#

import json
import string


class CCS811 : Driver

#- *************************************** -#   
    def init()
        light.set({"Power2":true, "rgb": "FF0000", "bri":255})
        tasmota.set_timer(1000)
        light.set({"Power2":true, "rgb": "000080", "bri":255})
        tasmota.set_timer(1000)
        light.set({"Power2":true, "rgb": "4FFF02", "bri":255})
        tasmota.set_timer(1000)
    end

#- *************************************** -# 
    def every_second()

        var sensors=json.load(tasmota.read_sensors())
        print(sensors)
         
        var CO2_Val=sensors['CCS811']['eCO2']
        var TVOC_Val=sensors['CCS811']['TVOC']
        print(sensors['BME280']['Temperature'])
        print('CO2:', CO2_Val, 'TVOC:', TVOC_Val)
        if CO2_Val < 800
            light.set({"Power2":true, "rgb": "4FFF02", "bri":255})
            print('Gruen ', CO2_Val)
        elif
            CO2_Val > 820 && CO2_Val < 1200
            light.set({"Power2":true, "rgb": "FFDC02", "bri":255})
            print('Gelb ', CO2_Val)
        else
            light.set({"Power2":true, "rgb": "FF0000", "bri":255})
            print('Rot ', CO2_Val)
        end
        
    end
    
    

end

#d1 = CCS811()
#tasmota.add_driver(d1)
tasmota.add_driver(CCS811())`

**Reply by sfromis** [reply]
Hard to say as you did not tell what error message you got. Still, those `tasmota.set_timer(1000)` are bad. It needs an argument with a function to be called after the interval, maybe something like:
`tasmota.set_timer(1000, /->light.set({"Power2":true, "rgb": "000080", "bri":255}))`

BTW. Do you **really** want a CO2 value of 810 to show up as red?
